pkg/build/generator/generator.go

```diff
@@ -211,6 +211,9 @@ func (g *BuildGenerator) checkLastVersion(bc *buildapi.BuildConfig, lastVersion
 // updateImageTriggers sets the LastTriggeredImageID on all the ImageChangeTriggers on the BuildConfig and
 // updates the From reference of the strategy if the strategy uses an ImageStream or ImageStreamTag reference
 func (g *BuildGenerator) updateImageTriggers(ctx kapi.Context, bc *buildapi.BuildConfig, from, triggeredBy *kapi.ObjectReference) error {
+	if triggeredBy == nil {
+		return nil
```

Comment:
Yes, maybe it shouldn't be `return nil`... I don't know what was the intended semantics, as it's the first time I'm touching this part of the code base.
I hope to know soon, though. I think @csrwng or @bparees might know better.

But if you look back at https://github.com/openshift/origin/pull/3748/files#diff-82f54c5f4bf72a64f869a69e557df8cbL121 you'll see that before that change, when the pointer was nil there was no error and we just proceeded. So the current implementation `return nil` keeps the old semantics.